### PR TITLE
Refactor and improve component API with f32 infinities

### PR DESCRIPTION
**Pull request by Nexfader**
First of all, thanks for the library! I wanted to improve its code quality, so I've decided to open this PR. Here's what I've done:

- Made use of Bevy's `Rect`, `Aabb2d` and Rust's `Range` types to make the code more idiomatic;
- Extracted some logic into separate functions, which reduced code duplication and also will be helpful in unit-testing later;
- Simplified bound handling: it now heavily uses `f32::INFINITY` and `f32::NEG_INFINITY`, making calculations seamless and simpler (as opposed to constantly dealing with `Option`s);
- Standardized the usage of `vec2` over `Vec2::new` (previously, the code contained both);
- Changed `single` to `get_single` in `camera_zoom` (similar to what #58 did for `camera_movement`);
- Reduced if-nesting using if-guards [[source](https://refactoring.guru/replace-nested-conditional-with-guard-clauses)];
- Formatted comments according to the [style guide](https://doc.rust-lang.org/nightly/style-guide/#comments).

Sorry if this turned out too opinionated. Feel free to discuss any points in the PR :)

### CODE REVIEW

**Comment by johanhelsing** [comment]
Looks great! Thanks!

I have some trouble understanding that one line, but otherwise this should be ready to merge :)

File: src/lib.rs [hunk 1]
```diff
@@ -57,182 +61,154 @@ fn check_egui_wants_focus(
...
+        let cursor_world_pos =
+            pos.translation.truncate() + cursor_normalized_screen_pos * proj_size * old_scale;
+        let proposed_cam_pos =
+            cursor_world_pos - cursor_normalized_screen_pos * proj_size * proj.scale;
+
+        // As we zoom out, we don't want the viewport to move beyond the provided
+        // boundary. If the most recent change to the camera zoom would move cause
+        // parts of the window beyond the boundary to be shown, we need to change the
+        // camera position to keep the viewport within bounds.
+        let aabb = cam.aabb().shrink(proj.area.size() / 2.);
```
[the remark this answers is not in the document]
Comment (Nexfader) [reply src/lib.rs]: Extracted `clamp_to_safe_zone` function. I really wanted replace `PanCam::aabb` with it, but I decided not to include implementation details in the component struct

File: src/lib.rs [hunk 2]
```diff
@@ -259,26 +236,65 @@ pub struct PanCam {
...
-    /// If present, the orthographic projection will be clamped to this boundary both
-    /// when dragging the window, and zooming out.
+    /// If present, the orthographic projection will be clamped to this boundary
+    /// both when dragging the window, and zooming out.
     pub max_y: Option<f32>,
 }
 
+impl PanCam {
+    /// Returns (min, max) bound tuple. If some bounds were not provided, either
+    /// `f32::INFINITY` or `f32::NEG_INFINITY` will be used.
```
[the remark this answers is not in the document]
Comment (Nexfader) [reply src/lib.rs]: Yeah, good idea! I've committed the rework

File: src/lib.rs [hunk 3]
```diff
@@ -305,12 +299,12 @@ impl Default for PanCam {
             grab_buttons: vec![MouseButton::Left, MouseButton::Right, MouseButton::Middle],
             enabled: true,
             zoom_to_cursor: true,
-            min_scale: 0.00001,
-            max_scale: None,
-            min_x: None,
-            max_x: None,
-            min_y: None,
-            max_y: None,
+            min_scale: f32::NEG_INFINITY,
```
Comment (johanhelsing) [inline src/lib.rs:302]: Was this an accidental change? How does a negative/zero scale behave?

Comment (Nexfader) [reply src/lib.rs]: Oops, you're right. I didn't test this properly😳. I've reverted the change

**Review by johanhelsing — APPROVED** [review]
Awesome. Great work :)

**Pull request merged**